http/src/error.rs

```diff
@@ -0,0 +1,28 @@
+#![allow(missing_docs)]
+
+use std::io;
+
+use hyper;
+
+error_chain! {
+	foreign_links {
+		Io(io::Error);
+	}
+
+	errors {
+		/// Hyper error
+		Hyper(e: hyper::Error) {
```

Comment:
why not `foreign_links`? It [implements](https://docs.rs/hyper/0.11.6/hyper/error/enum.Error.html#impl-StdError) `std::error::Error`

Reply:
Explanation given in custom `From` implementation. We want to unwrap `Io` error from hyper error.

Reply:
but why?